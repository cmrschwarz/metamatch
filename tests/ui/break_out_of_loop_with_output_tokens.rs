use metamatch::unquote;

fn main() {
    let res = unquote! {
        let mut X = 0;
        let super res = loop {
            X += 1;
            if X == 10 {
                break X;
            }
            quote!(X);
        };
        quote!(res);
    };
    assert_eq!(res, 10);
}
