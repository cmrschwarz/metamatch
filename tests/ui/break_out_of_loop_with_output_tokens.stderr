error: `break` value in `loop` with non empty output token list
 --> tests/ui/break_out_of_loop_with_output_tokens.rs:9:17
  |
9 |                 break X;
  |                 ^^^^^
